include/heap.h

```diff
@@ -3,81 +3,105 @@
    See LICENSE and COPYING for usage.
 
  *********************************************************************/
+#define MAX_BLOCK_SIZE 1600
+#define MAX_BLOCK_COUNT 16
 
 #define DECLARE_HEAP(type, orderby) \
     struct heap_ ## type {   \
         uint32_t size;    \
         uint32_t n;       \
-        type *top;        \
+        type *top[MAX_BLOCK_COUNT];        \
     }; \
     typedef struct heap_ ## type heap_ ## type; \
+    static inline type* getElement(struct heap_ ## type *heap, uint32_t idx) \
+    { \
+        uint32_t elements_per_block = MAX_BLOCK_SIZE/sizeof(type); \
```

Comment:
Considering that `elements_per_block` is completely known at compile-time, and most of the divisions happen with that variable, I expect the compiler to optimise this quite decently. It may cost us a few more cycles, but nothing huge. Also, it's not like adding timers is the inner loop of some heavy iterations or something.
But if you're really worried, feel free to do some measurements and prove me wrong :-)